### ISSUE TITLE
Compiler exception on namespace default export of class/function

### ISSUE THREAD

**Issue by zamb3zi** [Microsoft/TypeScript#17770]
**TypeScript Version:**  Playground (2.4?)

**Code**

```ts
namespace W {
    export default 123;
    // generates as expected: A default export can only be used in an ECMAScript-style module.
}

namespace W {
    export default function () { }; // alternatively: class {}
    // throws TypeError: Cannot read property 'pos' of undefined
}
```

**Expected behavior:**

The compiler should generate the same error in both cases rather than throwing a TypeError.

**Comment by ajafff** [comment]
Duplicate of #16301
Duplicate of #16300

Fixed by #16704

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
